src/main/java/com/novauc/controllers/HibernateQueryMethodsController.java

```diff
@@ -0,0 +1,100 @@
+package com.novauc.controllers;
+
+import com.novauc.entities.Customer;
+import com.novauc.services.CustomerRepository;
+import com.novauc.entities.Purchase;
+import com.novauc.services.PurchaseRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import javax.annotation.PostConstruct;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * Created by Merlin on 3/16/17.
+ */
+@Controller
+public class HibernateQueryMethodsController {
+    @Autowired
+    CustomerRepository customers;
+
+    @Autowired
+    PurchaseRepository purchases;
+
+//    @RequestMapping(path = "/", method = RequestMethod.GET)
+
+//    @RequestMapping(path = "/", method = RequestMethod.GET)
+//    public String home(Model model, String category) {
+//        List<Purchase> purchaseList = (ArrayList) purchases.findAll();
+//        if (category != null) {
+//            purchaseList = (List) purchases.findByCategory(category);
+//        } else {
+//            purchaseList = (List) purchases.findAll();
+//        }
+//        model.addAttribute("purchases", purchaseList);
+//        model.addAttribute("category", category);
+//        return "home";
+//    }
+
+        @RequestMapping(path = "/", method = RequestMethod.GET)
+    public String home(Model model, String category, Integer page) {
+        page = (page == null) ? 0 : page;
+        PageRequest p_r = new PageRequest(page, 20);
+        Page<Purchase> pur;
```

Comment:
Great Job Ross.  Pages should be 5 per however you changed it so number doesn't really matter the fact that you changed it shows proficiency.